src/factorization.jl

```diff
@@ -303,6 +310,7 @@ end
 
 Base.@kwdef struct KLUFactorization <: AbstractFactorization
     reuse_symbolic::Bool = true
+    check::Bool = true
```

Comment:
```suggestion
    check::Bool = false
```

We don't want to check by default, since downstream algorithms want retcodes not errors.

Reply:
oh wait, this is different from singularity checking? Maybe `check_pattern` then?

Reply:
This is checking that pattern is the same yes.